### ISSUE TITLE
micrometer doesn't support `ExecutorServiceMetrics` for `DefaultBlockingTaskExecutor`

### ISSUE THREAD

**Issue by ghkim3221** [line/armeria#3710]
From #3389, the `DefaultBlockingTaskExecutor` is introduced as default blocking task executor. Unfortunately, because micrometer's [ExecutorServiceMetrics](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/ExecutorServiceMetrics.java#L291) binds `ExecutorService` using class name, the `DefaultBlockingTaskExecutor` is not supported. So the following warning message is logged every start up:
```
Failed to bind as com.linecorp.armeria.common.util.DefaultBlockingTaskExecutor is unsupported.
```

**Comment by minwoox** [comment]
Thanks for the report, @ghkim3221! I never expected that this could be a problem. 😅 
Yeah, I guess unwrapping the `DefaultBlockingTaskExecutor` in the `ServerConfig` and calling `ExecutorServiceMetrics.monitor()` with the `delegate` is the best we can do for now.

We might also leave a warning log when the `blockingTaskExecutor` is not an instance of `ThreadPoolExecutor` or `ForkJoinPool` in the `ServerConfig` before we call `ExecutorServiceMetrics.monitor()`

**Comment by minwoox** [comment]
Or we could probably ask Micrometer team to add an interface such as `DelegatingScheduledExecutorService` and make `DefaultBlockingTaskExecutor` implement it.

**Comment by ikhoon** [comment]
> calling ExecutorServiceMetrics.monitor() with the delegate is the best we can do for now.

+1

> an interface such as DelegatingScheduledExecutorService

Or `MonitorableExecutorService` 🤣

**Comment by ghkim3221** [comment]
> ExecutorServiceMetrics.monitor() with the delegate is the best we can do for now.

+1 too.

**Comment by ikhoon** [comment]
Fixed by #3718
